Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/ManipulatorManager.cpp

```diff
@@ -67,6 +67,11 @@ namespace AzToolsFramework
             return;
         }
 
+        if (m_activeManipulator && m_activeManipulator->GetManipulatorId() == manipulator->GetManipulatorId())
+        {
+            m_activeManipulator.reset();
+        }
```

Comment:
question: Have you checked this logic doesn't impact other manipulator behaviors like the ones we use for entities and other component modes? As this is a central change we just want to make sure there aren't any unintended consequences. Could you do a little adhoc testing around this (like losing focus with alt-tab while dragging a manipulator). I think there are some bugs related to this printing warnings in the log already.

Reply:
hmm, I think I can collect a list of places where this is used and we can go through each one :?

Reply:
What might be ideal just to sanity check things, is to create a test using the Manipulator Test Framework to simulate this behavior. It should be possible as far as I know and would just ensure that this is not going to have any negative consequences. Some good things to test would be when dragging a manipulator in the regular viewport (entities) and with say with the Spline Component, and see what happens if UnregisterManipulator is called before a mouse-up event (I think this happens when alt-tabbing as one example). A test would be even better though to try and simulate the problem you were seeing before and showing this change fixes it.

Reply:
Is there already a test case for the whitebox manipulator that I can start from? I'll look into the Manipulator TestFramework and see what I can come up with.

Reply:
There are lots of example in `EditorTransformComponentSelectionTests` and I'm certain we do also have some White Box Manipulator tests in the White Box Gem, they shouldn't be too hard to find I don't think. The best thing would be to inherit from ToolsApplicationFixture using one of the Manipulator Test Framework mixins. Let me know how you get on and I can try and point you in the right direction if you get stuck. Thanks for looking into this 👍

Reply:
looks like there is already a test case: SwitchToRestoreModeDestroysModiferWhileInteracting

Reply:
That's awesome thanks @pollend! 😄 Taking a look now 👍